moveit_planners/ompl/ompl_interface/src/ompl_interface.cpp

```diff
@@ -196,95 +237,74 @@ void ompl_interface::OMPLInterface::loadPlannerConfigurations()
+          continue;
         }
-        else
+
+        double value_d;
+        if (nh_.getParam(group_names[i] + "/" + KNOWN_GROUP_PARAMS[k], value_d))
         {
-          double value_d;
-          if (nh_.getParam(group_names[i] + "/" + KNOWN_GROUP_PARAMS[k], value_d))
-            specific_group_params[KNOWN_GROUP_PARAMS[k]] = boost::lexical_cast<std::string>(value_d);
-          else
-          {
-            int value_i;
-            if (nh_.getParam(group_names[i] + "/" + KNOWN_GROUP_PARAMS[k], value_d))
-              specific_group_params[KNOWN_GROUP_PARAMS[k]] = boost::lexical_cast<std::string>(value_i);
-            else
-            {
-              bool value_b;
-              if (nh_.getParam(group_names[i] + "/" + KNOWN_GROUP_PARAMS[k], value_b))
-                specific_group_params[KNOWN_GROUP_PARAMS[k]] = boost::lexical_cast<std::string>(value_b);
-            }
-          }
+          specific_group_params[KNOWN_GROUP_PARAMS[k]] = boost::lexical_cast<std::string>(value_d);
+          continue;
+        }
+
+        int value_i;
+        if (nh_.getParam(group_names[i] + "/" + KNOWN_GROUP_PARAMS[k], value_d))
+        {
+          specific_group_params[KNOWN_GROUP_PARAMS[k]] = boost::lexical_cast<std::string>(value_i);
+          continue;
+        }
+
+        bool value_b;
+        if (nh_.getParam(group_names[i] + "/" + KNOWN_GROUP_PARAMS[k], value_b))
+        {
+          specific_group_params[KNOWN_GROUP_PARAMS[k]] = boost::lexical_cast<std::string>(value_b);
+          continue;
         }
       }
     }
 
-    // set the parameters (if any) for the default group configuration;
-    if (!specific_group_params.empty())
+    // add default planner configuration
+    planning_interface::PlannerConfigurationSettings default_pc;
+    std::string default_planner_id;
+    if (nh_.getParam(group_names[i] + "/default_planner_config", default_planner_id))
+    {
+      if (!loadPlannerConfiguration(group_names[i], default_planner_id, specific_group_params, default_pc))
+        default_planner_id = "";
+    }
+    if (default_planner_id.empty())
     {
-      planning_interface::PlannerConfigurationSettings pc;
-      pc.name = group_names[i];
-      pc.group = group_names[i];
-      pc.config = specific_group_params;
-      pconfig[pc.name] = pc;
+      default_pc.group = group_names[i];
+      default_pc.config = specific_group_params;
+      default_pc.config["type"] = "geometric::RRTConnect";
     }
+    default_pc.name = group_names[i];  // this is the name of the default config
+    pconfig[default_pc.name] = default_pc;
 
     // get parameters specific to each planner type
     XmlRpc::XmlRpcValue config_names;
     if (nh_.getParam(group_names[i] + "/planner_configs", config_names))
     {
-      if (config_names.getType() == XmlRpc::XmlRpcValue::TypeArray)
+      if (config_names.getType() != XmlRpc::XmlRpcValue::TypeArray)
       {
-        for (int32_t j = 0; j < config_names.size(); ++j)
```

Comment:
Part of it, yes.
I did this to make use of it to load the specified default planner above.